ratis-client/src/main/java/org/apache/ratis/client/impl/DataStreamClientImpl.java

```diff
@@ -44,6 +48,7 @@
 
   private DataStreamClientRpc dataStreamClientRpc;
   private OrderedStreamAsync orderedStreamAsync;
+  private final ClientId clientId;
```

Comment:
The clientId and the groupId/group should be passed in the constructor.  Let's add a TODO
```suggestion
  // TODO Similar to RaftClientImpl, pass ClientId and RaftGroupId/RaftGroup in constructor.
  private final ClientId clientId = ClientId.randomId();
  private final RaftGroupId groupId =  RaftGroupId.randomId();
```